mythril/laser/ethereum/state.py

```diff
@@ -115,22 +106,15 @@ def mem_extend(self, start, size):
         :param start: Start of memory extension
         :param size: Size of memory extension
         """
-        if start < 4096 and size < 4096:
-
-            if size and start + size > len(self.memory):
-                n_append = start + size - len(self.memory)
 
-                while n_append > 0:
-                    self.memory.append(0)
-                    n_append -= 1
+        if size and start + size > len(self.memory):
+            n_append = start + size - len(self.memory)
 
-                # FIXME: this does not seem right
-                self.memory_size = size
+            while n_append > 0:
+                self.memory.append(0)
+                n_append -= 1
```

Comment:
we can write it as self.memory += n_append*[0] . It would be simple.

Reply:
Fixed, and added a nice test